### ISSUE TITLE
Improve toggle buttons accessibility

### ISSUE THREAD

**Issue by afercia** [WordPress/health-check#264]
The toggle buttons used across the interface, for example:
- Show passed tests
- Show all/fewer plugins
- Show all/fewer themes

need some improvements for better usability and accessibility:
- don't remove or hide them once activated, otherwise focus is lost
- use `aria-expanded` to communicate their collapsible panels state

**Comment by afercia** [comment]
For the plugins/themes ones, the best option would be to use just _one_ button.

**Comment by afercia** [comment]
Re: the plugins / themes buttons, it's important to note they can't be placed at the bottom of the list.

These buttons appear when there are more than 5 themes or 5 plugins. Each toggle is made of two buttons ("All" and "Fewer") and their visibility gets toggled. This produces a focus loss that should be fixed using only one button.

Regardless, when a toggle gets activated, the expanded "thing" should be placed _after_ the toggle and not before otherwise the revealed content would be "skipped" for keyboard users and assistive technologies users.

<img width="736" alt="screenshot 2019-02-24 at 15 10 41" src="https://user-images.githubusercontent.com/1682452/53300422-53b95880-3847-11e9-9ad7-52ec7fe747be.png">

A better pattern for accessibility would be:
- activate the toggle
- `aria-expanded` communicates something appeared in the page
- at this point, users expects to find some new content _after_ the toggle, not before

**Comment by Clorith** [comment]
Sounds good, would it work to change the text of the one button based on the state, or would that cause some confusion for screen readers?

Or, even better, would it help if the headings, "Available themes" had a button that said "Toggle all" alongside it? Of course then with a screen reader text, I'm thinking something along the lines of "There are X themes available, only the first Y are shown, toggle the full list". This would ensure the button comes first, and provides context to assistive tech early on. The downside is that you'd have to go through the first 5 entries (or however many the limit is set to) to reach the toggled ones, but if the toggle option comes first, it may be natural to make the judgement call there and then to toggle them or not?

**Comment by Clorith** [comment]
Aha, you were thinking along the same lines, I just noticed the PR reference (I missed it the first time around).

I'm still curious if my thoughts on the screen reader text might be an idea to follow through with though as context is, in my mind at least, important to know if you want to go through a potentially much larger list, and if you rely on a screen reader you'll want to know that you are now extending the data to go over by a potentially large amount and it's not as easy for them to "undo" that?

**Comment by afercia** [comment]
Thanks for reviewing :)

Well, when entering a list with a screen reader, the number of items in the list gets announced. This helps users understand how long the list is in advance. Safari and VoiceOver have a bug (or a feature...) and need an explicit `role="list"`. Added in the last commit.

<img width="662" alt="screenshot 2019-02-25 at 21 14 57" src="https://user-images.githubusercontent.com/1682452/53366056-06baac80-3943-11e9-9425-9cb185174637.png">

As per the design changes, yep the toggle could be changed to buttons, as long as the aria-label makes sense and there's an accessible tooltip to expose the label (this could be a bit trickier). I'd rather use buttons with visible text, as they're usable by everyone. Maybe making the font smaller and/or placing the toggle buttons on the right. This part needs design feedback and I guess it can be addressed separately :)

**Comment by Clorith** [comment]
Yeha I was thinking of placing the toggle button to the right of the title, and if the list count is exposed we don't need a lavish extra text :)

**Comment by Clorith** [comment]
This was changed in #272, we'll tackle any design related changes to the toggle style in a new ticket when applicable.
